### PR TITLE
Documentation Changes

### PR DESCRIPTION
**Pull request by Titan-C**
This is to showcase a bit more Sphinx-Gallery, putting a short version of the gallery on the front page of the documentation.
I also imported some instructions on how to use Sphinx-Gallery from my talk at CDS

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,7 +10,9 @@ Getting Started to Sphinx-Gallery
     :alt: Documentation Status
 
 
-Sphinx extension for automatic generation of an example gallery.
+A Sphinx extension that builds an HTML version of any python
```
Comment (GaelVaroquaux) [inline README.rst]: Complete nitpick, but the 'Python' language takes a capital 'P'.

File: README.rst [hunk 2]
```diff
@@ -25,8 +27,9 @@ You can do a direct install via pip by using::
 Install as developer
 --------------------
 
-You can get the latest development source from our Github repository.
-You need  ``setuptools`` installed in your system to install.
+You can get the latest development source from our Github
+repository. You need ``setuptools`` installed in your system to
```
Comment (GaelVaroquaux) [inline README.rst]: You should put a link here.

File: doc/advanced_configuration.rst [hunk 3]
```diff
@@ -5,24 +5,10 @@ Advanced Configuration
 Here are the personal configurations that you can modify within Sphinx-Gallery.
 
 
-Changing default directories
-============================
```
Comment (GaelVaroquaux) [inline doc/advanced_configuration.rst]: This may be a very stupid question, but why is this removed?

File: doc/index.rst [hunk 4]
```diff
@@ -6,10 +6,45 @@
 Welcome to Sphinx-Gallery's documentation!
 ==========================================
 
-Sphinx extension for automatic generation of an example gallery.
+A Sphinx extension that builds an HTML version of any python
+script and puts it into an examples gallery.
```
Comment (GaelVaroquaux) [inline doc/index.rst]: Suggested rephrasing:

A sphinx (link to sphinx homepage)  extension that builds an HTML gallery of examples from any set of Python scripts.

File: doc/index.rst [hunk 5]
```diff
@@ -6,10 +6,45 @@
...
+
 It is extracted from the scikit-learn project and aims to be an
 independent general purpose extension.
 
+Why do we need it?
```
Comment (GaelVaroquaux) [inline doc/index.rst]: Suggested rephrasing:

"Why sphinx-gallery?"

File: doc/index.rst [hunk 6]
```diff
@@ -6,10 +6,45 @@
...
+------------------
+
+* Because we have a lot of code that does something good enough for
+  someone to learn about it.
+* Looking to the visual output is simpler to our minds than reading
+  the source code.
+* The Gallery workflow:
+   * Search for something that more or less looks to what you want
+   * Learn about it
```
Comment (GaelVaroquaux) [inline doc/index.rst]: Suggested rephrasing:

* Simple examples that run out of the box are the best way to learn a library
* Pleasing, organized, visual layouts
* Links, searching, backlinks throughout examples and documentation

File: doc/index.rst [hunk 7]
```diff
@@ -6,10 +6,45 @@
...
+
+How does it look like
```
Comment (GaelVaroquaux) [inline doc/index.rst]: "What do it look like?"

File: doc/index.rst [hunk 8]
```diff
@@ -6,10 +6,45 @@
...
+---------------------
+
+You can structure your gallery from your python scripts and embed it
+in any place of your website. And it look like the next sample.
```
Comment (GaelVaroquaux) [inline doc/index.rst]: Suggested rephrasing (for the whole paragraph above):

Here is an example gallery generated from a few Python scripts.

File: doc/index.rst [hunk 9]
```diff
@@ -6,10 +6,45 @@
...
+------------------
+
+* Because we have a lot of code that does something good enough for
+  someone to learn about it.
+* Looking to the visual output is simpler to our minds than reading
+  the source code.
+* The Gallery workflow:
+   * Search for something that more or less looks to what you want
+   * Learn about it
+
```
Comment (GaelVaroquaux) [inline doc/index.rst]: I would suggest to add a section here called 'Features' with the following content:

* Runs Python files and captures outputs + figures
* Manages an arbitrary number of galleries
* Automatically links back from code in the gallery to the corresponding function documentation
* Automatically links from documentation of functions to examples using the function
* Notebook-style examples: turn your scripts into a full narrative

Where each of the items should have a link to a page of the documentation showcasing the item and explaining how to do it.

**Comment by GaelVaroquaux** [comment]
Overall, this looks great and is a strong improvement.

I made a lot of comments on the front page, because it is very important, and I felt that even nitpicks could be useful.

**Comment by GaelVaroquaux** [comment]
By the way, travis fails on your PR, and it seems legit.

File: doc/advanced_configuration.rst (hunk 3 above)
Comment (Titan-C) [inline doc/advanced_configuration.rst]: Is not really removed.  It's only moved to the main readme. Making part of the standard configuration to declare where your examples directories are.

File: doc/index.rst (hunk 7 above)
Comment (Titan-C) [inline doc/index.rst]: Thank you had no idea about this one
